ignite/contrib/handlers/trains_logger.py

```diff
@@ -0,0 +1,578 @@
+                                 event_name=Events.EPOCH_COMPLETED)
+
+            # Attach the logger to the trainer to log optimizer's parameters, e.g. learning rate at each iteration
+            trains_logger.attach(trainer,
+                                 log_handler=OptimizerParamsHandler(optimizer),
+                                 event_name=Events.ITERATION_STARTED)
+
+            # Attach the logger to the trainer to log model's weights norm after each iteration
+            trains_logger.attach(trainer,
+                                 log_handler=WeightsScalarHandler(model),
+                                 event_name=Events.ITERATION_COMPLETED)
+
+
+    """
+
+    def __init__(self, *_, **kwargs):
+        try:
+            import trains
+        except ImportError:
+            raise RuntimeError(
+                "This contrib module requires trains to be installed. "
+                "You may install trains using: \n pip install trains \n"
+            )
+
+        experiment_kwargs = {
+            k: v for k, v in kwargs.items() if k not in (
+                "project_name", "task_name", "task_type", "report_freq",
+                "histogram_update_freq_multiplier", "histogram_granularity"
+            )
+        }
+
+        self.task = trains.Task.init(
+            project_name=kwargs.get("project_name"),
+            task_name=kwargs.get("task_name"),
+            task_type=kwargs.get("task_type", trains.Task.TaskTypes.training),
+            **experiment_kwargs
+        )
+
+        self.trains_logger = self.task.get_logger()
+
+        self.grad_helper = trains.binding.frameworks.tensorflow_bind.WeightsGradientHistHelper(
+            logger=self.trains_logger,
+            report_freq=kwargs.get("report_freq", 100),
+            histogram_update_freq_multiplier=kwargs.get("histogram_update_freq_multiplier", 10),
+            histogram_granularity=kwargs.get("histogram_granularity", 50),
+        )
+
+
+class TrainsSaver(DiskSaver):
+    """Handler that saves input checkpoint as Trains artifacts
+
+    Examples:
+
+        .. code-block:: python
+
+            from ignite.contrib.handlers.trains_logger import *
+            from ignite.handlers import Checkpoint
+
+            trains_logger = TrainsLogger(project_name="pytorch-ignite-integration",
+                                         task_name="cnn-mnist"
+                                         )
+
+            to_save = {"model": model}
+
+            handler = Checkpoint(to_save, TrainsSaver(trains_logger), n_saved=1,
+                                 score_function=lambda e: 123, score_name="acc",
+                                 filename_prefix="best",
+                                 global_step_transform=global_step_from_engine(trainer))
+
+            validation_evaluator.add_event_handler(Events.EVENT_COMPLETED, handler)
+
+    """
+
+    def __init__(self, logger: TrainsLogger, output_uri=None, dirname=None, *args, **kwargs):
```

Comment:
Nit: would you mind removing the type for `logger` or adding types to the remaining args? It's not necessary for the whole class, but consistency within a single method would be nice. 😃

Reply:
Sure 😄